### DISCUSSION TITLE
screen resolution

### DISCUSSION THREAD

**Discussion by swiatmar** [NightscoutFoundation/xDrip#1932] [Q&A]
Hi
I nstalled xdrip on my android watch. and somehow it is cutted. It means i can see the ap only on the half screen. there is some problem with the resolution. can i edit it somehow? The watch ist Kospet optimus 2 and has an resolution 360x360 px.

**Comment by tolot27** [comment]
Is it running Wear OS or the full Android version and wich version?

**Comment by swiatmar** [comment]
It is a full android watch Kospet Optimus2 with android 10.7
This is how it looks like:
https://discourse.fullandroidwatch.org/uploads/default/optimized/3X/e/2/e2dc0697284689ae89523c6103075f0805e85843_2_375x500.jpeg

**Comment by tolot27** [comment]
@parapenT1sta Does your friend sees a similar issue his LEMFO LEMP?

In general, xDrip and full Android is designed to run on devices having a rectangle display whereas xDrip wear and Wear OS are designed to run on rectangle and circular/rounded displays. It would be a huge effort to adapt the xDrip Android app to circular displays.

**Comment by parapenT1sta** [comment]
@tolot27 the watch I was testing had the option to switch from rounded to square mode. In my case I was using it in  square mode and only switch to round when sometimes was hard to press in some options.

**Reply by swiatmar** [reply]
my watch has also the option to switch to square. but the problem is that the xdrip doesnt appear on the whole screen . it is pushed up in both version round and square like you can see on the picture

**Reply by swiatmar** [reply]
@parapenT1sta  which android version did you have on your watch? And does the xdrip appear on the whole screen?  because in my case it is pushed up and the bottom is empty.

**Reply by parapenT1sta** [reply]
@swiatmar I don't have the watch anymore, just had it to test but I remember it had Android 9.1 but for it work better it also had an option beside the one to switch from Circle to Square. In the watch I didn't have the option to select the font size, check if yours have it and select font "small".
In the watch applications, there was an option called: "Optimization" and I had to select:
>> Third-Party apps adapter > Turn ON ( I )

I saved in a TXT all the steps I had to perform to make it work perfectly fine:
ANDROID SYSTEM SETTINGS

> Apps & Notifications
>> xDrip+ > Permissions: Set All: On; Data Usage: Set Both Options: On

> Battery
>> Battery Manager > OFF;
>> Battery Saver: Smart Saving Mode; Low Power Save: OFF; Schedule power Saving: OFF;
>> App standby optimizer:
>>> xDrip+ > App Optimized: OFF; Standby wakeup optimization: Do Not Optimize; Standby sleep optimization: Do Not Optimize; Standby network data: Do Not Optimize
>> Auto Launch management > App Auto Launch: xDrip: ON; App secondary launch: xDrip: ON

> Location
>> Use Location: ON
>> App Level permissions: Set all ON (make sure xDrip and Google Play Services are ON)
>> Advanced > Scanning > Bluetooth Scanning: ON; Google Location Accuracy: ON

APPLICATIONS (WATCH)

> Optimization
>> Clean Task > Turn OFF ( o )
>> Third-Party apps adapter > Turn ON ( I )

**Reply by swiatmar** [reply]
@parapenT1sta. ah okay. well like I said i have problem with wrong displaying the app not the font size etc. I can turn also fron round to square have font size settings. The problem is that xdrip are completly pushed up like an app         this is round[]https://discourse.fullandroidwatch.org/uploads/default/optimized/3X/e/2/e2dc0697284689ae89523c6103075f0805e85843_2_375x500.jpeg / and this is square []https://photos.app.goo.gl/VwinCcb1D6L4EgSm8  really strange. I have also one with android 9 and there is everything okay
